### PR TITLE
Refactor main controller

### PR DESCRIPTION
**Pull request by mauroz77**
- Add Test for AjaxController
- Apply google code style to AjaxController
- Delete ZOOMA related unused code.
- Format MappingsService code with Google code style
- Delete rearchArchive method in MappingService because it is obsolete.
- Rename AjaxController to MappingsController
- Rename old MappingController to UserController
- Set controllers path as /api/mappings for all endpoints in MappingsController
- Rename endpoint /api/getmissingdiagnosismappings to /api/mappings/getmissingmappings
- Delete obsolete endpoint /api/mappings/archive

### CODE REVIEW

File: rest/src/main/java/org/pdxfinder/controllers/AjaxController.java [hunk 1]
```diff
@@ -252,90 +236,6 @@ public Object exportMappingData(HttpServletResponse response,
     return csvReport;
   }
 
-
-  /****************************************************************
-   *                   INTERACTIONS WITH ZOOMA                    *
-   ****************************************************************/
```
Comment (Afollet) [inline rest/src/main/java/org/pdxfinder/controllers/AjaxController.java]: Was this the duplication you had mentioned?

**Review by Afollet — APPROVED** [review]
Looks good. Feel free to delete any of those sections of commented out code if you feel they are unnecessary.

I'm glad to you have you on the team.

File: rest/src/main/java/org/pdxfinder/controllers/AjaxController.java (hunk 1 above)
Comment (mauroz77) [reply rest/src/main/java/org/pdxfinder/controllers/AjaxController.java]: That is code for endpoints that are not being used in the UI, and they didn't work well either, so they are obsolete.

File: rest/src/main/java/org/pdxfinder/util/JsonHelper.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    DateFormat df = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss a z");
+    objectMapper.setDateFormat(df);
+  }
+
+  private JsonHelper() {}
+
+  /**
+   * Parses the given JSON string into a Java object using a standard Jackson mapper.
+   *
+   * @param json the JSON string to parse.
```
[the remark this answers is not in the document]
Comment (mauroz77) [reply rest/src/main/java/org/pdxfinder/util/JsonHelper.java:29]: I removed this for consistency.

File: rest/src/test/java/org/pdxfinder/controllers/AjaxControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+  @BeforeEach
+  void setUp() {
+  }
+
+  @AfterEach
+  void tearDown() {
+  }
+
+  @Test
+  public void testGetMappings() throws Exception {
```
[the remark this answers is not in the document]
Comment (mauroz77) [reply rest/src/test/java/org/pdxfinder/controllers/AjaxControllerTest.java]: I changed the names to use the long format for consistency and if we decide to make the names short later I will change these.

**Pull request merged**